q2_quality_control/tests/test_quality_control.py

```diff
@@ -46,19 +50,106 @@ def _load_DNAFASTAFormat(reads_fn):
         self.fungal_ref = _load_DNAFASTAFormat('fungal-ref-sequences.fasta')
         self.fungal_exp = _dnafastaformats_to_series(
             _load_DNAFASTAFormat('fungal-query-sequences.fasta'))
-
-    def test_exclude_seqs_blast(self):
+        self.query_seqs_with_mismatch = _load_DNAFASTAFormat(
+            'query-sequences-with-mismatch.fasta')
+        self.query_seqs_short = _load_DNAFASTAFormat(
+            'query-sequences-short.fasta')
+        self.query_seqs_part_rand = _load_DNAFASTAFormat(
+            'query-partially-random.fasta')
+
+    def test_exclude_seqs_bacterial_hit_fungal_miss(self):
+        for method in ['blast', 'vsearch']:
```

Comment:
In these tests where you're iterating over the methods, do you know which method was used when you get a test failure? If not, that could be annoying to debug later on. If the method isn't reported, you could include that in the [optional `msg` parameter to `assertEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual), or I could give you some other suggestions for how to structure this to avoid duplicating the tests (e.g., you could have a base class with the tests, and then override that in subclasses that have a `method` variable that is used in all of the tests).

Reply:
@gregcaporaso thanks for the suggestion about test subclasses — if it won't take long, could you offer some pseudocode to direct me as I implement this?

Reply:
👍